setup_yarn.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -e
+
+require_var() {
+  if [ -z "$1" ]; then
+    >&2 echo "$2"
+    return 1
+  fi
+  return 0
+}
+
+require_env() {
+  require_var "$2" "Missing env var '$1'"
+}
+
+setup_yarn() {
+
+  if [ "$ARTIFACTORY_YARN_ID_TOKEN" == "false" ]; then
+    echo "Skipping yarn setup because ARTIFACTORY_YARN_ID_TOKEN=$ARTIFACTORY_YARN_ID_TOKEN"
+    return 0
+  fi
+
+  require_env "ARTIFACTORY_YARN_ID_TOKEN" "$ARTIFACTORY_YARN_ID_TOKEN" || return 1
+
+  local yarnrc
+  yarnrc="$HOME/.yarnrc.yml"
+  cat > "$yarnrc" << EOF
+npmAlwaysAuth: true
+npmAuthToken: ${ARTIFACTORY_YARN_ID_TOKEN}
+
+EOF
+  echo "Wrote to $yarnrc"
+  return 0
+}
+
+if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
+  if setup_yarn "$@"; then
```

Comment:
Isn't line 38-41 the same as just calling `setup_yarn "$@"` alone? The `set -e` on top means the script will exit with its exit status if it wasn't 0.

Reply:
honestly i have no idea what that part does (starting from line 37), i just made the new script as similar to the others as possible to keep maintainability

we should ask Mark why did he use it in the others before

Reply:
Ah okay I didn't realize this was based on some existing scripts. It could be simplified as I indicated, I think, but I'll leave that up to you.

Reply:
I have found that Bash does not exit in the context of an `if` statement even when using `set -e`


```shell
set -e

if ls /wefwefwe; then
  echo "Exists"
else
  echo "Does not exist"
fi

echo "Got to the end"
```

shows

```
/tmp/exit_or_no.sh
ls: /wefwefwe: No such file or directory
Does not exist
Got to the end
```

Reply:
That would be correct; the `if` uses the exit status to determine whether the conditional is true or not.

Your example would match the code in question here better if it looked like this:

```bash
#!/usr/bin/env bash

set -e

if true; then
  if ls $1; then
    exit 0
  else
    exit 1
  fi
fi

echo "Got to the end"
```

Which returns the same exit code as my above proposal, which would make the sample code look like this:

```bash
#!/usr/bin/env bash

set -e

if true; then
  ls $1
  echo "Exists"
fi

echo "Got to the end"
```

```
$ /tmp/test_foo /tmp/non-existent; echo $?
ls: /tmp/non-existent: No such file or directory
1

$ /tmp/test_foo /tmp; echo $?
test_foo

Exists
Got to the end
0
```

Reply:
👍